net/socket/socket.h

```diff
@@ -38,6 +41,27 @@ class NET_EXPORT Socket {
   virtual int Read(IOBuffer* buf,
                    int buf_len,
                    CompletionOnceCallback callback) = 0;
+#if defined(STARBOARD)
+  struct ReadPacketResult {
+    char* buffer;
+    int result;
+  };
+  struct ReadPacketResults {
+    void clear() {
+      buffer = nullptr;
+      result = 0;
+      packet_buffer_size = 0;
+      packets = nullptr;
+    }
+    scoped_refptr<IOBufferWithSize> buffer;
+    int result = 0;
+    int packet_buffer_size = 0;
+    ReadPacketResult* packets = nullptr;
+  };
+  virtual int ReadMultiplePackets(ReadPacketResults* results,
```

Comment:
Idk if it's practical to make this a pure virtual ? Then you don't need a default impl in `socket.cc`

Reply:
This intentionally has a default implementation that returns "not implemented". There are many classes that implement this API, but we only implement the function for direct UDP connections (UDPSocketStarboard).

If I make it a pure virtual, then all other classes that inherit this need an implementation.

Reply:
ack